### ISSUE TITLE
Nodes have gaps

### ISSUE THREAD

**Issue by Gummibeer** [DurtyFree/gta-v-data-dumps#56]
Hey,

at the very first: thanks for this awesome repository! 🙏
It helps so much to create tools around GTA/FiveM/...

If I draw all the connected nodes from [nodes.json](https://github.com/DurtyFree/gta-v-data-dumps/blob/master/nodes.zip) on a map there are a ton of gaps - which makes real navigation impossible as nearly all roads have a gap somewhere in the middle.
Is there any reason behind it? Or does GTA solve that on runtime by finding the closest node if it reaches one of these dead ends and "jumps" to this one? 🤔
So for sure I can go through the nodes and add all the missing pieces or find an algorithm to add connections between these dead-ends. Just wanted to check if there's an "official" way or if the dump misses some connections between single nodes.

![Bildschirmfoto 2022-10-20 um 11 10 30](https://user-images.githubusercontent.com/6187884/196909120-4cb23a41-999b-4f43-b504-82f50f2a6df4.png)
![Bildschirmfoto 2022-10-20 um 11 19 18](https://user-images.githubusercontent.com/6187884/196909329-5230304e-d6c2-4ac3-9a3b-72a8adff21e3.png)
![Bildschirmfoto 2022-10-20 um 11 19 36](https://user-images.githubusercontent.com/6187884/196909358-6fe48ad3-e0a5-4130-87c3-40c95f79d0a9.png)

**Comment by DurtyFree** [comment]
I guess this all makes somehow sense for their path finding algorythm - could also be related to various flags that were unknown when I first time looked into this. Some flags indicate for example whether a node is only valid for special cases, or for lane switching, turning or whatever. Could all be related to this

**Comment by Gummibeer** [comment]
Yeah, the flags make sense - as it's nearly always related to larger junctions, multi-lane streets and driveways.
I'm nearly done connecting all of them, luckily I just need undirected connections.
As I'm pretty sure that directed connections would be even more missing. It was ~4000 deadends (nodes with just one connection) and also a lot of just missing connections.
